pkg/client/errors.go

```diff
@@ -147,3 +150,20 @@ type ObjectMissing struct{}
 func (e ObjectMissing) Error() string {
 	return "Object key is missing, object key cannot be empty"
 }
+
+// UnexpectedEOF (EPIPE) - reader closed prematurely.
+type UnexpectedEOF struct {
+	TotalWritten int64
+	TotalSize    int64
+}
+
+func (e UnexpectedEOF) Error() string {
+	return fmt.Sprintf("Data written ‘%d’ is smaller than the size ‘%d’ of the input Reader.", e.TotalWritten, e.TotalSize)
+}
+
+// UnexpectedExcessRead - reader wrote more data than requested.
+type UnexpectedExcessRead UnexpectedEOF
+
+func (e UnexpectedExcessRead) Error() string {
+	return fmt.Sprintf("Data written ‘%d’ is bigger than the size ‘%d’ of the input Reader.", e.TotalWritten, e.TotalSize)
```

Comment:
Received excess data from the input reader. Expected only ‘%d’ bytes, but received ‘%d’ bytes.